nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/UnpackContent.java

```diff
@@ -202,11 +213,11 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         final boolean addFragmentAttrs;
         switch (packagingFormat) {
             case TAR_FORMAT:
```

Comment:
Have you considered doing some restructuring of the _onTrigger()_ method?
Since the packaging format is based on the processor wide configuration property (not FlowFile) most of the logic (switch/case) could be performed once instead of every time _onTrigger()_ is called. That includes the new logic with regex pattern compile.
Or am I missing something big here?